android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -145,6 +148,21 @@ protected void onReceiveResult(int resultCode, Bundle resultData) {
           Intent packageIntent = new Intent(getApplicationContext(), PackageActivity.class);
           packageIntent.putExtras(bundle);
           startActivity(packageIntent);
+
+          // Determine if associated lexical model should be downloaded
+          if (FileUtils.hasKeymanPackageExtension(kmpFilename) && !FileUtils.hasLexicalModelPackageExtension(kmpFilename)
```

Comment:
When does this get called? Does it get called for .kmp downloaded from cloud as well as adhoc mechanisms?

Reply:
This path is reached after the DownloadIntentService downloads a kmp file. (The path from KMPBrowserActivity)

ad-hoc mechanisms are triggered by different code paths (file managers)